public/store/LogTypeStore.ts

```diff
@@ -42,6 +48,16 @@ export class LogTypeStore {
         };
       });
 
+      ruleTypes.splice(
+        0,
+        ruleTypes.length,
+        ...logTypes.map((logType) => ({
```

Comment:
Clarification question: Is the spread operator necessary here?

Reply:
Yes, as per the syntax [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice)